target/linux/ramips/image/mt7621.mk

```diff
@@ -860,6 +860,25 @@ define Device/xiaomi_mir3g
 endef
 TARGET_DEVICES += xiaomi_mir3g
 
+define Device/xiaomi_mir4
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+  KERNEL_SIZE := 4096k
+  IMAGE_SIZE := 124416k
+  UBINIZE_OPTS := -E 5
+  IMAGES += kernel1.bin rootfs0.bin
+  IMAGE/kernel1.bin := append-kernel
+  IMAGE/rootfs0.bin := append-ubi | check-size
```

Comment:
That's quite uncommon, isn't it?

Reply:
mir4  and mir3g are basically the same configuration, but without USB interface, RAM 256MiB →128MiB. So I used mir3g's compilation options